src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/ListingGenerationController.java

```diff
@@ -321,8 +321,23 @@ public ResponseEntity<ListingCallbackResponse> generateHearingDocument(
 
         var listingData = listingRequest.getCaseDetails().getCaseData();
         var caseTypeId = listingRequest.getCaseDetails().getCaseTypeId();
-
-        return getResponseEntity(listingData, caseTypeId, userToken);
+        List<String> errorsList = new ArrayList<>();
+        boolean invalidCharsExist = listingService.checkInvalidCharsForAllParties(
+                listingRequest.getCaseDetails(), userToken, errorsList);
+        if (hasListings(listingData) && !invalidCharsExist) {
```

Comment:
switch these two conditions round. !invalidCharsExist should come first for performance reasons. If hasListings did loads of work then it would be a waste of time if invalidCharsExist = true

Reply:
switched them